pkg/rebuild/pypi/infer.go

```diff
@@ -66,14 +80,33 @@ func (Rebuilder) InferRepo(t rebuild.Target, mux rebuild.RegistryMux) (string, e
 		if strings.Contains(url, "sponsors") {
 			continue
 		}
-		c, err := uri.CanonicalizeRepoURI(url)
-		if err == nil {
-			return c, nil
+		repoLinks = append(repoLinks, url)
+	}
+	// Four priority levels:
+	// 1. project source link | known repo
+	for _, url := range repoLinks {
+		if uri.SmellsLikeARepo(url) {
```

Comment:
instead of having this helper function, why not just return the outcome of Find and, if non-empty, pass that to Canonicalize?